src/transform/src/predicate_pushdown.rs

```diff
@@ -591,14 +606,25 @@ impl PredicatePushdown {
                             .collect::<Vec<_>>();
 
                         // For every single-input expression `expr`, we can push
-                        // down `expr = literal` and remove `expr` from the
+                        // down `expr = <runtime constant>` and remove `expr` from the
                         // equivalence class.
                         for (expr_pos, input, expr) in single_input_exprs.drain(..).rev() {
-                            push_downs[input].push(gen_literal_equality_pred(expr));
+                            push_downs[input].extend(gen_literal_equality_preds(expr));
                             equivalences[equivalence_pos].remove(expr_pos);
                         }
+
+                        // If none of the expressions in the equivalence depend on input
+                        // columns and equality predicates with them are pushed down,
+                        // we can safely remove them from the equivalence.
```

Comment:
We could probably add a TODO and/or follow-up issue for:
> However, perhaps what we should do is to push (1 = mz_logical_timestamp()) to all join inputs instead, to prevent any computation from happening until that condition is true...